### PR TITLE
Improve cancelation policy, add more periods and validations for them…

### PR DESCRIPTION
**Pull request by maptuhec**
…, for creating and canceling reservation

### CODE REVIEW

File: src/services/blockchain/hotelReservation.js [hunk 1]
```diff
@@ -68,9 +68,10 @@ export class HotelReservation {
 		);
 
 
-		await approveContract(wallet, reservationCostLOC, HotelReservationFactoryContract.address, gasPrice);
+		let approve = await approveContract(wallet, reservationCostLOC, HotelReservationFactoryContract.address, gasPrice);
```
Comment (Perseverance) [inline src/services/blockchain/hotelReservation.js]: Why do you need approve variable?

File: src/services/blockchain/validators/reservation-validators.js [hunk 2]
```diff
@@ -33,18 +36,37 @@ export async function validateReservationParams(jsonObj,
...
 		!hotelId ||
 		!roomId ||
 		!numberOfTravelers
 	) {
 		throw new Error(ERROR.INVALID_PARAMS);
 	}
+	if (daysBeforeStartForRefund.length != refundPercentages.length) {
+		throw new Error(ERROR.INVALID_REFUND_PARAMS_LENGTH);
+	}
+	if ((daysBeforeStartForRefund.length > maxRefundPeriods) ||
```
Comment (thcrnk) [inline src/services/blockchain/validators/reservation-validators.js]: We have already checked if daysBeforeStartForRefund.length != refundPercentages.length. So we don't need to check if both of them are > maxRefundPeriods or < 0

File: src/services/blockchain/hotelReservation.js (hunk 1 above)
Comment (thcrnk) [inline src/services/blockchain/hotelReservation.js]: If you trigger both approve and reserve transactions without waiting first for the approve to be mined is it working correctly?

Comment (maptuhec) [reply src/services/blockchain/hotelReservation.js]: Yes, this works as expected

File: src/services/blockchain/validators/reservation-validators.js (hunk 2 above)
Comment (maptuhec) [reply src/services/blockchain/validators/reservation-validators.js]: As discussed we will leave this validation for readability.

**Pull request merged**